### DISCUSSION TITLE
Is it possible to get number of nodes in a cluster?

### DISCUSSION THREAD

**Discussion by YarShev** [dask/dask#7060] [Q&A] [answered]
Hi there,

It is assumed there is spun up dask cluster. Is there a way to get number of nodes used in a cluster and also some additional info regarding available resources?

Thanks in advance!

**Comment by mrocklin** [comment, marked as answer]
Try the client.scheduler_info() function

**Comment by YarShev** [comment]
Yes, I saw this function. Getting result from the function I can iterate workers and retrieve ip address of each one. Then find unique ip addresses and count their amount. But is there more simple way? Like in ray it looks as follows `len(ray.nodes())`.

**Reply by jsignell** [reply]
To get the number of workers you can do `len(client.scheduler_info()["workers"])` that information is also displayed when you do `print(client)`.

**Reply by YarShev** [reply]
I am interested not in number of workers, but in number of nodes in a cluster used.

**Reply by jsignell** [reply]
Ah I understand. I don't think there is a more straightforward way in dask, but that could be a nice convenience method if you care to add it.

**Reply by YarShev** [reply]
Okay, thanks for the answer.
On the method, I am not pretty good at Dask architecture to add corresponding method. If I become better at the architecture, I will try to do it :)

**Comment by mrocklin** [comment]
Host is included in scheduler_info.  I think that what you want is a line
of python code on the results of that call.

Note that Dask doesn't group workers by host in the same way that Ray
does.  You may be trying to do something that doesn't make a ton of sense.

**Reply by YarShev** [reply]
I only wanted to know a simple way to get number of nodes. I think, we figured out how to achieve it looking at the discussion above. @mrocklin , thanks a lot!
